pkg/roachprod/vm/aws/aws.go

```diff
@@ -727,9 +727,26 @@ func (p *Provider) Delete(l *logger.Logger, vms vm.List) error {
 	return g.Wait()
 }
 
-// Reset is part of vm.Provider. It is a no-op.
+// Reset is part of vm.Provider.
 func (p *Provider) Reset(l *logger.Logger, vms vm.List) error {
-	return nil // unimplemented
+	byRegion, err := regionMap(vms)
+	if err != nil {
+		return err
+	}
+	g := errgroup.Group{}
+	for region, list := range byRegion {
+		args := []string{
+			"ec2", "reboot-instances",
+			"--region", region,
+			"--instance-ids",
+		}
+		args = append(args, list.ProviderIDs()...)
+		g.Go(func() error {
```

Comment:
Something to keep in mind although I don't think it's really relevant in this situation... Each cloud API imposes its own limits. Sometimes we may have to bound the number of concurrent requests. Since we don't really anticipate restarting 100s of nodes at a time, we should in theory be within those limits.